### DISCUSSION TITLE
BORG 1.9.04

### DISCUSSION THREAD

**Discussion by mikeberger** [mikeberger/borg_calendar#138] [General]
Throwing out a quick pre-release because I added lots of new themes (Java look and feels), and some look much better than the current ones. Many dark themes are also included. Java themes can be changed via the Look and Feel pull-down in the appearance options. The calendar colors must be changed separately via the User Color Scheme tab. Unfortunately, Themes that can be created under the User Color Scheme tab are unrelated to the Java Look and Feel "because reasons".

BORG 1.9.04
-----------
Add more themes (look-and-feel). Support Dark themes. #137
Add past due todos to reminder email #135
Fix where the "For repeating ToDo's, only show the current instance" option is loaded from when displaying the options UI. #134
completing tasks in aCalendar+ tasks (Android) doesn't properly update BORG for repeating todos #133
display a notification if delete or delete one only is pressed in the appt editor and no appts are selected #115
popup a notification if search returns no results #112
add option to only show the current todo for a repeating todo #106 - fix

<hr /><em>This discussion was created from the release <a href='https://github.com/mikeberger/borg_calendar/releases/tag/1.9.04'>BORG 1.9.04</a>.</em>

**Comment by ghost** [comment]
I would like to thank you for your great work. I have been watching this project for a very long time and I think there is a huge potential behind it.
 
I would like to work with it so much, but there are two important points that prevent me: 
- The address book should syncronize with Carddav. This is very important to have the same address book on all devices.
- Tasks should also sync. 

I use Baikal Server and DAV5x/Etar/Tasks on Android but there is nothing on the computer. I use Windws7. Earlier there was Lightning, but that has been destroyed by the Mozilla company.

I would like to help, so please describe your development environment. I'm programming PHP obiect oriented, and I've been squinting over to Java for a long time. Now it says Java Swing. This confuses me to some extent :) I would like to use VB code since I am used to it, so the question is which compiler could I use under Windows7 together with VB code?

**Reply by mikeberger** [reply]
Rather than replying in one super long post, I'll break up my responses into separate replies. You cover a lot of topics.

> `The address book should synchronize with Carddav. This is very important to have the same address book on all devices

I don't remember why I never did this. It may have been because I have few contacts myself. Or maybe my Address Book record is too different from the CardDav model. I'll add a github task to look into this.

#140 created

**Reply by mikeberger** [reply]
> Tasks should also sync.

Clarifying what events, todos, and tasks are is worth starting a separate discussion.

created discussion #142 . Todos can sync with android, given the right software on android.

> I use Baikal Server and DAV5x/Etar/Tasks on Android

I use Baikal. On Android I used CalDAV-Sync, which builds a tasks database that can be read by its own Open Tasks app and also by aCalendar+ - a great calendar app.

**Reply by mikeberger** [reply]
> I would like to help, so please describe your development environment. I'm programming PHP obiect oriented, and I've been squinting over to Java for a long time. Now it says Java Swing. This confuses me to some extent :) I would like to use VB code since I am used to it, so the question is which compiler could I use under Windows7 together with VB code?

BORG is written in Java. The UI elements are written using the Swing UI libraries that are part of Java. VB code cannot be added to BORG. To add features to BORG, you would need to learn Java.

BORG can be developed and run on any OS that supports Java. On windows, you need version 15 or higher of the Java Development Kit (JDK). I use openJDK. For an IDE, I use Eclipse.
